dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Augmentations/ToolsAugmentation.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Microsoft.Bot.Builder;
+using Microsoft.Teams.AI.AI.Models;
+using Microsoft.Teams.AI.AI.Planners;
+using Microsoft.Teams.AI.AI.Prompts;
+using Microsoft.Teams.AI.AI.Prompts.Sections;
+using Microsoft.Teams.AI.AI.Tokenizers;
+using Microsoft.Teams.AI.AI.Validators;
+using Microsoft.Teams.AI.State;
+
+namespace Microsoft.Teams.AI.AI.Augmentations
+{
+    /// <summary>
+    /// A server-side 'tools' augmentation
+    /// </summary>
+    public class ToolsAugmentation : IAugmentation
+    {
+        /// <summary>
+        /// Creates a plan given validated response value.
+        /// </summary>
+        /// <param name="context">Context for the current turn of conversation.</param>
+        /// <param name="memory">Interface for accessing state variables.</param>
+        /// <param name="response">Response to validate.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The created plan</returns>
+        public Task<Plan?> CreatePlanFromResponseAsync(ITurnContext context, IMemory memory, PromptResponse response, CancellationToken cancellationToken = default)
+        {
+            List<IPredictedCommand> commands = new();
+
+            if (response.Message != null && response.Message.ActionCalls != null && response.Message.ActionCalls.Count() > 0)
+            {
+                IList<ActionCall> actionCalls = response.Message.ActionCalls;
+
+                foreach (ActionCall actionCall in actionCalls)
+                {
+                    PredictedDoCommand command = new(actionCall.Function.Name, actionCall.Function.Arguments)
```

Comment:
GitHub is noting a bunch of check warnings here. Could you clean those up before check in please? Ty~

Reply:
fixed